include/seqan3/alignment/pairwise/detail/policy_affine_gap_with_trace_recursion.hpp

```diff
@@ -0,0 +1,133 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::policy_affine_gap_with_trace_recursion.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/pairwise/detail/policy_affine_gap_recursion.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Implements the alignment recursion function for the alignment algorithm using affine gap costs with trace
+ *        information.
+ * \ingroup pairwise_alignment
+ * \copydetails seqan3::detail::policy_affine_gap_recursion
+ */
+template <typename alignment_configuration_t>
+class policy_affine_gap_with_trace_recursion : protected policy_affine_gap_recursion<alignment_configuration_t>
+{
+protected:
+    //!\brief The type of the base policy.
+    using base_t = policy_affine_gap_recursion<alignment_configuration_t>;
+
+    // Import base types.
+    using typename base_t::traits_type;
+    using typename base_t::score_type;
+    using typename base_t::affine_score_tuple_t;
+
+    //!\brief The trace type to use.
+    using trace_type = typename traits_type::trace_type;
+    //!\brief The internal tuple storing the trace directions of an affine cell.
+    using affine_trace_tuple_t = std::tuple<trace_type, trace_type, trace_type>;
+    //!\brief The affine cell type returned by the functions.
+    using affine_cell_type = affine_cell_proxy<std::pair<affine_score_tuple_t, affine_trace_tuple_t>>;
+
+    // Import base member.
+    using base_t::gap_extension_score;
+    using base_t::gap_open_score;
+    using base_t::first_row_is_free;
+    using base_t::first_column_is_free;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    policy_affine_gap_with_trace_recursion() = default; //!< Defaulted.
+    policy_affine_gap_with_trace_recursion(policy_affine_gap_with_trace_recursion const &) = default; //!< Defaulted.
+    policy_affine_gap_with_trace_recursion(policy_affine_gap_with_trace_recursion &&) = default; //!< Defaulted.
+    policy_affine_gap_with_trace_recursion &
+    operator=(policy_affine_gap_with_trace_recursion const &) = default; //!< Defaulted.
+    policy_affine_gap_with_trace_recursion &
+    operator=(policy_affine_gap_with_trace_recursion &&) = default; //!< Defaulted.
```

Comment:
In `policy_affine_gap_with_trace_recursion_banded` you break the line differently. I don't care either way but it should be consistent at least throughout this code :)